File: ios/Classes/FlutterIgolfViewerPlugin.m
#import "FlutterIgolfViewerPlugin.h"
#if __has_include(<flutter_igolf_viewer/flutter_igolf_viewer-Swift.h>)
#import <flutter_igolf_viewer/flutter_igolf_viewer-Swift.h>
#else
#import "flutter_igolf_viewer-Swift.h"
#endif

@implementation FlutterIgolfViewerPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftFlutterIgolfViewerPlugin registerWithRegistrar:registrar];
}
@end

File: ios/Classes/FlutterIgolfViewerPlugin.swift
import Flutter
import UIKit

public class FlutterIgolfViewerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_igolf_viewer", binaryMessenger: registrar.messenger())
    let instance = FlutterIgolfViewerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
